internal/testplanet/planet.go

```diff
@@ -0,0 +1,210 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information
+
+// Package testplanet implements the full network wiring for testing
+package testplanet
+
+import (
+	"context"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
+	"storj.io/storj/internal/memory"
+	"storj.io/storj/pkg/auth/grpcauth"
+	"storj.io/storj/pkg/pb"
+	pieceserver "storj.io/storj/pkg/piecestore/rpc/server"
+	"storj.io/storj/pkg/piecestore/rpc/server/psdb"
+	"storj.io/storj/pkg/pointerdb"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/pkg/utils"
+	"storj.io/storj/storage/teststore"
+)
+
+// Planet is a full storj system setup.
+type Planet struct {
+	directory string // TODO: ensure that everything is in-memory to speed things up
+
+	nodeInfos []pb.Node
+	nodeLinks []string
+
+	nodes        []*Node
+	Satellites   []*Node // DO NOT MODIFY
+	StorageNodes []*Node // DO NOT MODIFY
+	Uplinks      []*Node // DO NOT MODIFY
```

Comment:
how come?

Reply:
Ah, previously modifying the slice would've caused memory leaks. The comments were stale.